### PR TITLE
Add memory hotplug/unplug support

### PR DESCRIPTION
**Pull request by xutian**
Add memory/pc-dimm device support in avocado-vt

### CODE REVIEW

**Comment by xutian** [comment]
deps on https://github.com/avocado-framework/avocado-vt/pull/218

**Comment by xutian** [comment]
deps on https://github.com/avocado-framework/avocado-vt/pull/194

**Comment by suqinhuang** [comment]
could you provide id

**Comment by suqinhuang** [comment]
LGTM
Acked-by: Suqin Huang

**Comment by spcui** [comment]
Acked-by: scui@redhat.com

**Comment by xutian** [comment]
deps on PR https://github.com/avocado-framework/avocado-vt/pull/230

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -1625,3 +1625,47 @@ def pcic_by_params(self, name, params):
...
+        dev.set_param("id", "%s-%s" % ("mem", name))
+        return dev
+
+    def dimm_device_define_by_params(self, name, params):
+        """
+        Create pc-dimm device from params.
+        """
+        params = params.object_params("dimm")
+        attrs = qdevices.Dimm.__attributes__[:]
+        params = params.copy_from_keys(attrs)
```
Comment (CongSmile) [inline virttest/qemu_devices/qcontainer.py]: Is there is default value for the params?

Comment (xutian) [inline virttest/qemu_devices/qcontainer.py]: No, just copy exits params if key exits in old params, not default params.

**Comment by xutian** [comment]
deps on PR  https://github.com/avocado-framework/avocado-vt/pull/247

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Pull request merged**